llvm/lib/Target/RISCV/RISCVCodeGenPrepare.cpp

```diff
@@ -155,6 +163,43 @@ bool RISCVCodeGenPrepare::visitIntrinsicInst(IntrinsicInst &I) {
   return true;
 }
 
+bool RISCVCodeGenPrepare::expandVPStrideLoad(IntrinsicInst &II) {
+  if (ST->hasOptimizedZeroStrideLoad())
+    return false;
```

Comment:
RISCVISelDAGToDAG.cpp will turn a splat of a scalar load back to a zero-strided load anyway if `ST->hasOptimizedZeroStrideLoad()`, I think we might be able to remove this check. The riscv_masked_strided_load lowering that does a splat of a scalar load doesn't seem to check for it anyway.

Reply:
I think I agree with you somewhat, lets leave it in. The undo-ing in RISCVISelDAGToDAG is a bit hidden, better to be explicit here.